client/main/lib/ClientForRemote.ts

```diff
@@ -0,0 +1,96 @@
+import DatastoreApiClient from '@ulixee/datastore/lib/DatastoreApiClient';
+import ConnectionParameters from './ConnectionParameters';
+import { IInputFilter, IOutputSchema } from '../interfaces/IInputOutput';
+import ILocationStringOrObject from '../interfaces/ILocationStringOrObject';
+
+export default class ClientForRemote {
+  public user: string;
+  public password: string;
+  public host: string;
+  public port: number;
+  public database: string;
+
+  #apiClient: DatastoreApiClient;
+
+  constructor(uriOrObject: ILocationStringOrObject) {
+    const connectionParameters = new ConnectionParameters(uriOrObject)
+    this.user = connectionParameters.user
+    this.database = connectionParameters.database
+    this.port = connectionParameters.port
+    this.host = connectionParameters.host
+    this.password = connectionParameters.password  
+  }
+
+  private get apiClient(): DatastoreApiClient {
+    if (!this.#apiClient) {
+      const address = `${this.host}:${this.port}`;
+      this.#apiClient = new DatastoreApiClient(address);
+    }
+    return this.#apiClient;
+  }
+
+  public async run<
+    TInputFilter extends IInputFilter = IInputFilter,
```

Comment:
This typing is confusing me. Does it actually narrow types that are input or output? Seems to just say, this extends basically anything, and defaults to anything.

Reply:
It's just a placeholder. We need to download and support schemas from remote datastores.